### ISSUE TITLE
Dependency keys are not cleared on LRU eviction

### ISSUE THREAD

**Issue by vladar** [benjamn/optimism#497]
Looks like there is a minor "leak" in optimism even with LRU enabled: entries of `depsByKey` always stay in memory even after all optimism `Entries` corresponding to the given key are cleared.

So this `key` + an empty `dep` Set are "leaking":

https://github.com/benjamn/optimism/blob/4301eaca4eee1c29b80a14f65fde8683e4b144fe/src/dep.ts#L31-L35

In my testing it also survives Apollo eviction / GC, even `cache.gc({ resetResultCache: true })`

**Comment by vladar** [comment]
Some extensive testing in our app gave the following results:

```
JS Heap
Without the fix (mb): 218, 221, 213, 220, 220 (avg 218.4 mb)
With the fix (mb):    208, 210, 209, 208, 202 (avg 207.4 mb)
```

**Comment by vladar** [comment]
Benchmark and repro: https://github.com/vladar/optimism-deps-fix
